File: Python/Binary_Search.py
#Contributed by @Hinal-Srivastava
def binary_search(list,item):
    first = 0
    last = len(list)-1
    flag = False
    while( first<=last and not flag):
        mid = (first + last)//2
        if list[mid] == item :
            flag = True
            print("Element found at ", mid," after sorting the list")
        else:
            if (item < list[mid]):
                last = mid - 1
            else:
                first = mid + 1
    if(flag == False):
        print("Element not part of given list")            	
            
#Driver Program
ele_lst = [] 
n = int(input("Enter number of elements : ")) #Length of list
print("Enter Elements \n")

for i in range(n):
    element = int(input("\t"))
    ele_lst.append(element)

ele_lst.sort()
item_=int(input("Enter search element : "))

print(binary_search(ele_lst, item_))
    

File: RUBY/binarysearch.rb
def binary_search(an_array, item)
    first = 0
    last = an_array.length - 1

    while first <= last
        i = (first + last) / 2

        if an_array[i] == item
            return "#{item} found at position #{i}"
        elsif an_array[i] > item
            last = i - 1
        else an_array[i] < item
            first = i + 1
        end
    end

    return "#{item} not found in this array"
end